src/main/java/com/google/sps/servlets/CreateQuestionServlet.java

```diff
@@ -0,0 +1,149 @@
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.*;
+import java.util.Date;
+import javax.servlet.ServletException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+/** Servlet that creates and stores questions
+* @author Klaudia Obieglo 
+*/
+@WebServlet("/createQuestion")
+public class CreateQuestionServlet extends HttpServlet{
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /* Servlet Receives information from the client about the question they want to save */
+    Long date = (new Date()).getTime();
+    String question = getParameter(request, "question", "");
+    String marks = getParameter(request, "marks", "");
+    
+    UserService userService = UserServiceFactory.getUserService();
+    String ownerID = userService.getCurrentUser().getEmail(); 
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Create a Question Entity with the parameters provided
+    Entity questionEntity = new Entity("Question");
+    questionEntity.setProperty("question",question);
+    questionEntity.setProperty("marks",marks);
+    questionEntity.setProperty("date",date);
+    questionEntity.setProperty("ownerID",ownerID);
+    try{
+      datastore.put(questionEntity);    
+    }catch (DatastoreFailureException e){
+      System.out.println("Datastore is not responding right now. Try Again Later");
+    }
+    addQuestionToExamList(questionEntity.getKey().getId(),ownerID);
+
+    response.sendRedirect("/createQuestion.html");
+    response.setContentType("application/json");
+    response.getWriter().println(convertToJsonUsingGson(questionEntity));
+  }
+  private void addQuestionToExamList(long questionEntityKey,String ownerID)
+  {
+    /*Function that adds the question id to the list of questions in the exam entity
+    * Arguments : QuestionEntityKey - id of the question Entity we are adding to the list
+    *           : ownerID - email of the person who is adding this question to their exam
+    */
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    //grab the latest exam created by the user
+    Entity latestExam = getExam(ownerID);
```

Comment:
I don't think getting a latest exam created by a user is a good way to find the exam for the question. Imagine a situation when a user has 2 tabs open creating 2 exams in parallel. With current approach it's likely that first exam's questions will be attributed to the second exam.

Reply:
Fixed it, now theres a drop down where the latest test created by the user is default but if a user wants to add a qs to a different test they can do so my clicking on the drop down and changing it